From: John Hubbard <jhubbard@nvidia.com>
Subject: selftests/mm: kvm, mdwe fixes to avoid requiring "make headers"
Date: Fri, 7 Jun 2024 19:10:21 -0700

On Ubuntu 23.04, the kvm and mdwe selftests/mm build fails due to missing
a few items that are found in prctl.h.  Here is an excerpt of the build
failures:

ksm_tests.c:252:13: error: use of undeclared identifier 'PR_SET_MEMORY_MERGE'
...
mdwe_test.c:26:18: error: use of undeclared identifier 'PR_SET_MDWE'
mdwe_test.c:38:18: error: use of undeclared identifier 'PR_GET_MDWE'

Fix these errors by adding the missing items to vm_util.h, and include
vm_util.h from mdwe_test.c.

Link: https://lkml.kernel.org/r/20240608021023.176027-4-jhubbard@nvidia.com
Signed-off-by: John Hubbard <jhubbard@nvidia.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 tools/testing/selftests/mm/mdwe_test.c |    1 +
 tools/testing/selftests/mm/vm_util.h   |   15 +++++++++++++++
 2 files changed, 16 insertions(+)

--- a/tools/testing/selftests/mm/mdwe_test.c~mm-selftests-kvm-mdwe-fixes-to-avoid-requiring-make-headers
+++ a/tools/testing/selftests/mm/mdwe_test.c
@@ -15,6 +15,7 @@
 #include <unistd.h>
 
 #include "../kselftest_harness.h"
+#include "vm_util.h"
 
 #ifndef __aarch64__
 # define PROT_BTI	0
--- a/tools/testing/selftests/mm/vm_util.h~mm-selftests-kvm-mdwe-fixes-to-avoid-requiring-make-headers
+++ a/tools/testing/selftests/mm/vm_util.h
@@ -61,3 +61,18 @@ unsigned long get_free_hugepages(void);
 
 #define PAGEMAP_PRESENT(ent)	(((ent) & (1ull << 63)) != 0)
 #define PAGEMAP_PFN(ent)	((ent) & ((1ull << 55) - 1))
+
+#ifndef PR_SET_MEMORY_MERGE
+#define PR_SET_MEMORY_MERGE		67
+#endif
+
+#ifndef PR_GET_MEMORY_MERGE
+#define PR_GET_MEMORY_MERGE		68
+#endif
+
+#ifndef PR_SET_MDWE
+#define PR_SET_MDWE			65
+#define PR_MDWE_REFUSE_EXEC_GAIN	(1UL << 0)
+#define PR_MDWE_NO_INHERIT		(1UL << 1)
+#define PR_GET_MDWE			66
+#endif
_
